File: backend/src/main/java/com/tameem/pricewatch/dto/TrackedProductResponse.java
package com.tameem.pricewatch.dto;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

public record TrackedProductResponse(
        Long id, String name, String brand, String category,
        BigDecimal targetPrice, Instant createdAt, String imageUrl,
        String currency, BigDecimal currentPrice, int storeCount, boolean targetPriceReached,
        List<BigDecimal> recentPrices , BigDecimal trendPercent,
        List<String> stores
) {}

File: frontend/src/types/TrackedProductResponse.ts
export interface TrackedProductResponse {
  id: number;
  name: string;
  brand: string | null;
  category: string | null;
  targetPrice: number | null;
  createdAt: string;
  imageUrl: string | null;
  currency: string | null;
  currentPrice: number | null;
  storeCount: number;
}
